drv/Lists.drv

```diff
@@ -46,6 +46,16 @@ public final class LISTS {
 		}
 		return l;
 	}
+	
+	public static KEY_GENERIC LIST KEY_GENERIC reverse(final LIST KEY_GENERIC l) {
+		int size = l.size();
+		for (int i=0, mid=size>>1, j=size-1; i<mid; i++, j--) {
```

Comment:
This should be done with a view I think (see, e.g. Guavas Lists#reverse) for `O(1)` reversal instead of `O(n)`.

Reply:
Guavas implementation is a Lazy implementation, with a ObjectAllocation where you lose the original List Reference.
That just adds a layer on top of the existing one.
This is the direct implementation and should be O(1) or O/2 since it stops at the half point.

Also yes a lazy implementation can make sense if you don't want to modify the original, but the intend of this method is to modify the original. Could be compared to List.sort using Lists.sort where it gives you a sorted list that does not modify the original.

Reply:
This implementation is `O(n)` :-) 

The problem is that this method assumes the given list is random access (which may not always be the case). At least some sort of warning-documentation should be added then.

Reply:
How about we implement non random access list in the first place?
A warning may be a solution but then again why not implement it into the interface and then in to the list implementation?
Could you please decide?

Reply:
I don't see what implementing non-random access lists has to do with this discussion.

In case you want an in-place swap, I suggest using Java's `Collection#reverse` as starter. They have a good implementation which also works in linear time for properly implemented non-random access lists (instead of quadratic as this implementation would). However, fastutil doesn't use the `RandomAccess` markers I think. So it should be noted in the javadoc that this method assumes the given list to be random access.

I don't think it is sensible to add this to the list interface. I don't see any significant benefit.